### ISSUE TITLE
KotlinGenerator produces compilation errors in UDT Records when <recordsImplementingRecordN/> is true

### ISSUE THREAD

**Issue by mel4tonin4** [jOOQ/jOOQ#10923]
### Expected behavior

Build succesfully.

### Actual behavior

Failure to build with `mvn clean install` with:
```
[ERROR] Failed to execute goal org.jetbrains.kotlin:kotlin-maven-plugin:1.4.20-RC:compile (compile) on project jooq-mcve: Compilation failure: Compilation failure: 
[ERROR] /mnt/live/Development/3rdparty/jOOQ/jOOQ-mcve-udt-field-failure/src/main/kotlin/org/jooq/mcve/kotlin/udt/records/MyUdtRecord.kt:[36,49] Unresolved reference: ID
[ERROR] /mnt/live/Development/3rdparty/jOOQ/jOOQ-mcve-udt-field-failure/src/main/kotlin/org/jooq/mcve/kotlin/udt/records/MyUdtRecord.kt:[37,48] Unresolved reference: VALUE
```

### Steps to reproduce the problem

Build this fork of jOOQ-mcve: [https://github.com/mel4tonin4/jOOQ-mcve-udt-field-failure](https://github.com/mel4tonin4/jOOQ-mcve-udt-field-failure)

### Versions

- jOOQ: 3.14.3
- Java: 11.0.9, vendor: Ubuntu, runtime: /usr/lib/jvm/java-11-openjdk-amd64
- Database (include vendor): PostgreSQL
- OS: Ubuntu 20.04.1 LTS
- JDBC Driver (include name if inofficial driver): org.postgresql:postgresql:42.2.9

Notes: at the moment, it's not a problem with me because I'm not using the Kotlin code generator in my project, so I just disabled it in order to create the MCVE of my actual issue.

**Comment by lukaseder** [comment]
Thanks a lot for your report and for bearing with me on this issue :)

I'll look into this immediately

**Comment by lukaseder** [comment]
I can confirm, your MCVE produces these compilation errors:

```
[ERROR] Failed to execute goal org.jetbrains.kotlin:kotlin-maven-plugin:1.4.20-RC:compile (compile) on project jooq-mcve: Compilation failure: Compilation failure:
[ERROR] C:\Users\lukas\jOOQ\mcve\udts\jOOQ-mcve-udt-field-failure\src\main\kotlin\org\jooq\mcve\kotlin\udt\records\MyUdtRecord.kt:[36,49] Unresolved reference: ID
[ERROR] C:\Users\lukas\jOOQ\mcve\udts\jOOQ-mcve-udt-field-failure\src\main\kotlin\org\jooq\mcve\kotlin\udt\records\MyUdtRecord.kt:[37,48] Unresolved reference: VALUE
```

**Comment by lukaseder** [comment]
A workaround is to turn off `<recordsImplementingRecordN>false</recordsImplementingRecordN>`, which does not produce the offending code

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.4 (#10924). Thanks again for your report!

**Comment by mel4tonin4** [comment]
Thank you for fixing the issue!
